runtime/rasdump/javadump.cpp

```diff
@@ -1587,8 +1606,9 @@ outerMemCategoryCallBack (U_32 categoryCode, const char * categoryName, UDATA li
 	 * for all children of this node.
 	 */
 	memset(&total, 0, sizeof(memcategory_total));
-	total.category_bitmask = (U_32*)alloca(writer->_TotalCategories * sizeof(U_32));
-	memset(total.category_bitmask, 0, writer->_TotalCategories * sizeof(U_32));
+	U_32 bitmask_size = (((writer->_MaxCategoryBits - 1) / 32) + 1) * sizeof(U_32);
```

Comment:
This would be safer written like this (so no one has to wonder whether `_MaxCategoryBits` can be zero):
```
U_32 bitmask_size = ((writer->_MaxCategoryBits + 31) / 32) * sizeof(U_32);
```

Reply:
updated